java/vector/src/main/java/org/apache/arrow/vector/complex/ListViewVector.java

```diff
@@ -0,0 +1,945 @@
+        "ListViewVector does not support getReader operation yet.");
+  }
+
+  /**
+   * Get the size (number of bytes) of underlying buffers used by this
+   * vector.
+   * @return size of underlying buffers.
+   */
+  @Override
+  public int getBufferSize() {
+    if (valueCount == 0) {
+      return 0;
+    }
+    final int offsetBufferSize = valueCount * OFFSET_WIDTH;
+    final int sizeBufferSize = valueCount * SIZE_WIDTH;
+    final int validityBufferSize = getValidityBufferSizeFromCount(valueCount);
+    return offsetBufferSize + sizeBufferSize + validityBufferSize + vector.getBufferSize();
+  }
+
+  /**
+   * Get the size (number of bytes) of underlying buffers used by this.
+   * @param valueCount the number of values to assume this vector contains
+   * @return size of underlying buffers.
+   */
+  @Override
+  public int getBufferSizeFor(int valueCount) {
+    if (valueCount == 0) {
+      return 0;
+    }
+    final int validityBufferSize = getValidityBufferSizeFromCount(valueCount);
+
+    return super.getBufferSizeFor(valueCount) + validityBufferSize;
+  }
+
+  /**
+   * Get the field associated with the list view vector.
+   * @return the field
+   */
+  @Override
+  public Field getField() {
+    if (field.getChildren().contains(getDataVector().getField())) {
+      return field;
+    }
+    field = new Field(field.getName(), field.getFieldType(), Collections.singletonList(getDataVector().getField()));
+    return field;
+  }
+
+  /**
+   * Get the minor type for the vector.
+   * @return the minor type
+   */
+  @Override
+  public MinorType getMinorType() {
+    return MinorType.LIST;
+  }
+
+  /**
+  * Clear the vector data.
+  */
+  @Override
+  public void clear() {
+    super.clear();
+    validityBuffer = releaseBuffer(validityBuffer);
+    lastSet = -1;
+  }
+
+  /**
+  * Release the buffers associated with this vector.
+  */
+  @Override
+  public void reset() {
+    super.reset();
+    validityBuffer.setZero(0, validityBuffer.capacity());
+    lastSet = -1;
+  }
+
+  /**
+   * Return the underlying buffers associated with this vector. Note that this doesn't
+   * impact the reference counts for this buffer, so it only should be used for in-context
+   * access. Also note that this buffer changes regularly, thus
+   * external classes shouldn't hold a reference to it (unless they change it).
+   *
+   * @param clear Whether to clear vector before returning, the buffers will still be refcounted
+   *              but the returned array will be the only reference to them
+   * @return The underlying {@link ArrowBuf buffers} that is used by this
+   *         vector instance.
+   */
+  @Override
+  public ArrowBuf[] getBuffers(boolean clear) {
+    setReaderAndWriterIndex();
+    final ArrowBuf[] buffers;
+    if (getBufferSize() == 0) {
+      buffers = new ArrowBuf[0];
+    } else {
+      List<ArrowBuf> list = new ArrayList<>();
+      list.add(offsetBuffer);
+      list.add(validityBuffer);
+      list.add(sizeBuffer);
```

Comment:
Isn't this the wrong order?

Reply:
I have doubts about it? Do you have a recommendation?

Reply:
Shouldn't it go validity, offset, size?